plugins/module_utils/disk.py

```diff
@@ -113,7 +146,11 @@ def from_ansible(cls, ansible_dict):
             cache_mode=ansible_dict.get("cache_mode", None),
             name=ansible_dict.get("iso_name", None),
             disable_snapshotting=ansible_dict.get("disable_snapshotting", None),
-            tiering_priority_factor=ansible_dict.get("tiering_priority_factor", None),
+            tiering_priority_factor=TIERING_PRIORITY_MAPPING_FROM_ANSIBLE[
+                ansible_dict["tiering_priority_factor"]
+            ]
+            if ansible_dict.get("tiering_priority_factor", None) is not None
+            else None,
```

Comment:
I think, as a rule, we are storing into our ansible classes the ansible-friendly values. So the tiering_priority_factor should contain numbers 0 to 11, and to HC3 API we send "0, 1, 2,... 1024, 10240".